retrofit/src/test/java/retrofit2/ResponseTest.java

```diff
@@ -54,6 +55,28 @@
     assertThat(response.body()).isNull();
   }
 
+  @Test public void successWithHeaders() {
+    Object body = new Object();
+    Response<Object> response = Response.success(body, Headers.of("foo", "bar"));
```

Comment:
I'd prefer if you put the `Headers` here in a local and did `isSameAs` just like is being done with `body`.